src/cmds/scripts/post_install.py

```diff
@@ -0,0 +1,463 @@
+print "Post install process started..."
+
+import sys, getopt
+import os 
+import subprocess
+import socket
+from tempfile import mkstemp
+from shutil import move
+from os import fdopen, remove
+import os.path
+import time
+
+install_path= ''
+pbs_path=''
+
+
+def replace(file_path, pattern, subst):
+    #Create temp file
+    #print file_path
+    fh, abs_path = mkstemp()
+    with fdopen(fh,'w') as new_file:
+        with open(file_path) as old_file:
+            for line in old_file:
+                new_file.write(line.replace(pattern, subst))
+    #Remove original file
+    remove(file_path)
+    #Move new file
+    move(abs_path, file_path)
+	
+	
+
+
+def validate_credentials(username, password):
+   #print 'Printing from func' , validate_credentials.__name__
+   #print 'Username :', username
+   #print 'Password :', password 
+   #print install_path
+	if (username != '' and password != '' ):
+	   with open(os.devnull, 'w') as fp:
+			theproc= subprocess.Popen(["pbs_account.exe", "-c", "-s", "-a", username, "-p", password], stdout=fp, stderr=fp)
+			theproc.communicate()
+			#print theproc.returncode
+			if theproc.returncode > 0:
+				print "Invalid credentials"
+				sys.exit(3)
+			else:
+				print "User Validation Success"
+	else :
+		print "Usage error: \n post_install.py -t <type> -u <username> -p <password>"
+		sys.exit(4)
+		
+def createPbsConfFile():
+	pbspath=install_path[:-8]
+	pbsconffile=pbspath+'pbs.conf'
+	#print pbsconffile
+	#print socket.gethostname()
+	file = open(pbsconffile,"w")
+	file.write("PBS_SERVER="+socket.gethostname()+"\r\n")
+	file.write("PBS_EXEC="+pbspath+"exec\r\n")
+	file.write("PBS_HOME="+pbspath+"home\r\n")
+	file.write("PBS_START_SERVER=1\r\n")
+	file.write("PBS_START_COMM=1\r\n")
+	file.write("PBS_START_MOM=1\r\n")
+	file.write("PBS_START_SCHED=1\r\n")
+	file.close()
+	print "pbs.conf file created"
+
+def run_pbsmkdirs():
+	#print 'Inside run_pbsmkdirs'
+	os.system('pbs_mkdirs')
+	print 'Securing and Recreating done...'
+	
+def installing_vcredist():
+	print 'Installing Visual C++ Redist..'
+	#print install_path
+	inst_redist='"'+install_path+'\\vcredist_x86.exe" /passive /norestart'
+	redist_path=install_path+"\\vcredist_x86.exe"
```

Comment:
You should use os.path.join() to construct ALL paths (not just this one). It will use the appropriate delimiter (back/forward slash) for the system you are on.